### PR TITLE
[RFR] AuthProvider

### PR DESCRIPTION
**Pull request by djhi**
Follow #2 

Requiring and specifying the `authClient` for each `SwitchPermissions` and `WithPermission` components could be cumbersome. That's why we also provided an `AuthProvider` component.

### CODE REVIEW

File: src/AuthProvider.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import withContext from 'recompose/withContext';
+
+const AuthProviderComponent = ({ authClient, children }) => cloneElement(children, { authClient });
+
+AuthProviderComponent.propTypes = {
+    children: PropTypes.node.isRequired,
+};
+
+export default withContext(
+    {
```
Comment (fzaninotto) [inline src/AuthProvider.js]: maybe you can inline this one?

File: package.json [hunk 2]
```diff
@@ -51,7 +51,8 @@
   "dependencies": {
     "babel-plugin-transform-runtime": "~6.23.0",
     "lodash.get": "~4.4.2",
-    "proptypes": "~1.0.0"
+    "proptypes": "~1.0.0",
+    "recompose": "^0.23.4"
```
Comment (jpetitcolas) [inline package.json]: Shouldn't it be a `~` to be consistent?

File: src/WithPermission.js [hunk 3]
```diff
@@ -27,7 +29,7 @@ export default class WithPermission extends Component {
     };
 
     async componentWillMount() {
-        const { authClient, children, record, resource, permissions: requiredPersmissions, exact } = this.props;
+        const { authClient, children, record, resource, value: requiredPersmissions, exact } = this.props;
```
Comment (jpetitcolas) [inline src/WithPermission.js]: s/requiredPersmissions/requiredPermissions/

Comment (jpetitcolas) [inline src/WithPermission.js]: Why renaming `permissions`? Name is more meaningful IMO.

**Review by jpetitcolas — CHANGES_REQUESTED** [review]

File: src/WithPermission.js (hunk 3 above)
Comment (djhi) [reply src/WithPermission.js]: To be consistent with the `SwitchPermissions` & `Permission` components

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**